pkg/cri/store/image/image.go

```diff
@@ -85,6 +86,14 @@ func (s *Store) Update(ctx context.Context, ref string) error {
 			return fmt.Errorf("get image info from containerd: %w", err)
 		}
 	}
+	// parsing ref consistently for refCache, including adding `docker.io/library` prefix
+	if !strings.HasPrefix(ref, "sha256:") {
```

Comment:
is `ref` here in the format of `import-2022-11-28@sha256:c8a01d8610c642f0e6b83af5f6185b3eed34e322354be5a515da1a21453ac966`, and we are prefixing it with `docker.io/library` by calling `docker.ParseDockerRef()`? 

If so then why would the prefix be `sha256` here? Shouldn't it be the image name?

Reply:
> is `ref` here in the format of `import-2022-11-28@sha256:c8a01d8610c642f0e6b83af5f6185b3eed34e322354be5a515da1a21453ac966`, and we are prefixing it with `docker.io/library` by calling `docker.ParseDockerRef()`? 

In my testing:
- Before restarting containerd, `import-2022-11-28@sha256:c8a01d8610c642f0e6b83af5f61` is added to ref cache map during `ctr import`.
- - And removing by `crictl rmi` is using `docker.io/library/import-2022-11-28@sha256:c8a01d8610c642f0e6b83af5f61`.
- After restarting containerd, the cache became `docker.io/library/import-2022-11-28@sha256:c8a01d8610c642f0e6b83af5f61`
- - removing by `crictl rmi` works.


> If so then why would the prefix be `sha256` here? Shouldn't it be the image name?

If the ref is started with `sha256:`, we should not add the prefix like `docker.io/library/sha@333xxx`. This is not correct.

Reply:
> If the ref is started with sha256:, we should not add the prefix like docker.io/library/sha@333xxx. This is not correct.

So the ref could be in the format of 

a) `sha256:c8a01d8610c642f0e6b83af5f6185b3eed34e322354be5a515da1a21453ac966`, or 
b) `import-2022-11-28@sha256:c8a01d8610c642f0e6b83af5f6185b3eed34e322354be5a515da1a21453ac966`,

and we only want to prefix with `docker.io/library/` in case a) above. Is my understanding correct? It makes sense if so.

Reply:
Yes.

Reply:
Sorry for the late reply. Yep makes sense.

Reply:
One more question, have you run unit tests image_test.go? Seems the unit tests there assert on the ref, and your change will change the ref by prefixing `docker.io`?

Example https://github.com/containerd/containerd/blob/f0a32c66dad1e9de716c9960af806105d691cd78/pkg/cri/store/image/image_test.go#L178

Reply:
Thanks for the reminder.
- I updated the test case.